plugins/snapshot/src/main.ts

```diff
@@ -291,9 +293,12 @@ class SnapshotMixin extends SettingsMixinDeviceBase<Camera> implements Camera {
                 const cp = this.currentPicture;
                 debounced.catch(() => {});
                 try {
-                    picture = await timeoutPromise(1000, debounced);
+                    picture = await (options.timeout ? timeoutPromise(options.timeout, debounced) : debounced);
                 }
                 catch (e) {
+                    if (options.timeout)
+                        this.debugConsole.debug(`Periodic snapshot took longer than ${options.timeout} seconds to retrieve, falling back to cached picture.`)
```

Comment:
@koush I was thinking this should maybe a `.log()` instead of `.debug()` since it might be helpful to people who don't have debug enabled to easily see why their snapshots are stale.

Reply:
I think I meant for it to be log actually, so sure